chapters/operatorsandexpressions.tex

```diff
@@ -41,7 +41,11 @@ \section{Operator Precedence and Associativity}\label{operator-precedence-and-as
 % Beware that the array construction operator, normally expressed as \lstinline!{ }! needs escaped braces inside \caption.
 % This isn't handled correctly by LaTeXML, as reported here:
 % - https://github.com/brucemiller/LaTeXML/issues/1377
+\ifpdf
 \caption{Operators in order of precedence from highest to lowest, as derived from the Modelica grammar in \cref{modelica-concrete-syntax}.  All operators are binary except the postfix operators and those shown as unary together with \emph{expr}, the conditional operator, the array construction operator \lstinline!\{ \}! and concatenation operator \lstinline![ ]!, and the array range constructor which is either binary or ternary.  Operators with the same precedence occur at the same table row.}\label{tab:operator-precedence}
+\else
+\caption{Operators in order of precedence from highest to lowest, as derived from the Modelica grammar in \cref{modelica-concrete-syntax}.  All operators are binary except the postfix operators and those shown as unary together with \emph{expr}, the conditional operator, the array construction operator \lstinline!{ }! and concatenation operator \lstinline![ ]!, and the array range constructor which is either binary or ternary.  Operators with the same precedence occur at the same table row.}\label{tab:operator-precedence}
+\fi
```

Comment:
Does the ifpdf not work inside the caption? It feels bad to duplicate so much text

Reply:
To me the goal was that this should be temporary, and it seems slightly easier to remove if an entire line.
Since it is related to fragility within caption-context I didn't want to add to that with more conditionals inside it.

But I haven't tried if it works.

Reply:
That caption line is very long. If I look at it online:
https://github.com/modelica/ModelicaSpecification/blob/57e159fbf01bd5fef085951e3e1c730faa8c6f7c/chapters/operatorsandexpressions.tex#L44
I don't even see the part that needs to be different for ifpdf without much side-wise scrolling. If ifpdf was in the middle of it, it would be much harder to notice it. So I agree with Hans that (given that it's a temporary fix) it is easier to see where the condition is (and easier to remove it later) if we have the whole caption included in ifpdf.

Reply:
Oh, I would have written it as:
```latex
 \caption{Operators in order of precedence from highest to lowest, as derived from the Modelica grammar in \cref{modelica-concrete-syntax}.
All operators are binary except the postfix operators and those shown as unary together with \emph{expr}, the conditional operator, the array construction operator
\ifpdf
\lstinline!\{ \}!
\else
\lstinline!{ }!
\fi
and concatenation operator \lstinline![ ]!, and the array range constructor which is either binary or ternary.
Operators with the same precedence occur at the same table row.}
\label{tab:operator-precedence} 
```
But I guess it's OK

Reply:
We have the guideline that one sentence on one line, so it hasn't even occurred to me to break-up the line. Maybe the guideline should be amended to account for the cases like that?

Reply:
Sounds reasonable to me to add the `\ifpdf` example above as a style guide example of when it makes good sense to break up a sentence across lines.

Reply:
The problem I have with a multi-line caption is that the nesting isn't clear from individual lines.
Obviously one could indent lines etc, but it soon becomes complicated - especially as the contents of caption is deemed fragile by LaTeX.
As far as I could see we do not currently have any such cases.

(A completely different solution would be to not write excessive long captions.)